components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -229,10 +231,9 @@ class SitePermissionsFeature(
             storage.findSitePermissionsBy(request.host)
         }
 
-        val prompt = if (shouldApplyRules(permissionFromStorage)) {
```

Comment:
I was think about simpler solution wdyt? `shouldApplyRules` can stay as it is now and just verify that the request is for autoplay as for now we always want to handle autoplay with global rules.

```kotlin
        val prompt = if (request.isForAutoplay() || shouldApplyRules(permissionFromStorage)) {
            handleRuledFlow(request, session)
        } else {
            handleNoRuledFlow(permissionFromStorage, request, session)
        }
```